RoboJackets/robocup-software
# Wall tactic

## Description
Creates wall tactic, which builds a wall of n robots between the ball and the goal.

## Associated Issue
https://app.clickup.com/t/v5cpe0

## Steps to test
### Test Case 1
1. Start & run sim
2. Move ball slowly around the field (either with slow left-click + drag or kicks under 6 m/s)
3. Kick ball toward goal

Expected result: 3 robots (ids 0-2) should move to form a wall and track the ball (relatively) smoothly. They should block the ball from being scored. They should also never enter the penalty box or touch its bounding lines.

### Test Case 2
1. Start & run sim
2. Change the number in the `init()` method of gameplay's `play/wall_ball.py` that controls how many robots are in the wall tactic
3. Repeat Test Case 1

Expected result: N robots should move now, same behavior as before.

### Known Issues
 - Moving the ball too quickly will cause the robots to scramble
 - Sim will occasionally crash, complaining that:

```
[gameplay_node-11]     self.move.face_point = self.face_point
[gameplay_node-11] AttributeError: 'Move' object has no attribute 'face_point'
```

This is not true; both move action and move skill have a face_point. Relaunch sim if this error pops up.
